File: frontend/src/types/override.ts
// Cocok dengan backend/app/schemas/override.py — jaga urutan & isi tetap sama.
export type OverrideTargetType = "forecast_result" | "reorder_recommendation";

export interface Override {
  id: string;
  target_type: OverrideTargetType;
  target_id: string;
  user_id: string;
  previous_value: Record<string, unknown> | null;
  new_value: Record<string, unknown>;
  reason: string;
  created_at: string | null;
}

export interface OverrideInput {
  target_type: OverrideTargetType;
  target_id: string;
  new_value: Record<string, unknown>;
  reason: string;
}

File: backend/app/schemas/override.py
"""
Pydantic schemas untuk endpoint overrides — docs/ARCHITECTURE.md §4/§5.
"""
from typing import Literal

from pydantic import BaseModel


class OverrideCreateRequest(BaseModel):
    # target_type dibatasi Literal → nilai lain otomatis 422 (validasi Pydantic).
    target_type: Literal["forecast_result", "reorder_recommendation"]
    target_id: str
    new_value: dict
    reason: str = ""  # divalidasi di service: OVERRIDE_REASON_REQUIRED bila kosong


class OverrideOut(BaseModel):
    id: str
    target_type: str
    target_id: str
    user_id: str
    previous_value: dict | None
    new_value: dict
    reason: str
    created_at: str | None = None
